numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1192,145 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class CAR(Distribution):
+    r"""
+    The Conditional Autoregressive (CAR) distribution is a special case of the multivariate
+    normal in which the precision matrix is structured according to the adjacency matrix of
+    sites. The amount of autocorrelation between sites is controlled by :math:`\alpha`. The
+    distribution is a popular prior for areal spatial data.
+
+    :param float or ndarray loc: mean of the multivariate normal
+    :param float alpha: autoregression parameter. For most cases, the value should lie
+        between 0 (sites are independent, collapses to an iid multivariate normal) and
+        1 (perfect autocorrelation between sites), but the specification allows for negative
+        correlations.
+    :param float tau: positive precision for the multivariate normal
+    :param numpy.ndarray or scipy.sparse.csr_matrix W: symmetric adjacency matrix where 1
+        indicates adjacency between sites and 0 otherwise
+    :param bool is_sparse: whether to use a sparse form of W in calculations (must be True is
+        W is a :class:`scipy.sparse.spmatrix`)
```

Comment:
Maybe we need to mention that currently, only supports 2-dimensional W.

Reply:
I think it will always only support 2-dimensional W. It's an adjacency matrix of sites. In a 1D problem, the matrix would be tri-diagonal with a diagonal 0 (1 for sites to left and right, 0 otherwise).

It's similar to how the covariance matrix K for Gaussian processes is always 2D (for single outputs).

Would you still like me to mention that W must be 2D?

Reply:
Sure. If you want to support only 2D inputs, we just need to mention so in the docstring and raise an error to ask users to make a feature request if they want a batch of matrices. I have seen several requests for batch GPs but supporting it would require us to rewrite a bunch of computations in pyro.contrib.gp. The situation here is simpler because the current code already almost works (modulo tests pass). In case you don't want to support batched W, just ignore the corresponding tests.

Reply:
What sort of occasions would a batched GP be useful? Multi-output?

When might a batched CAR be useful?

Reply:
Just think about this for simplicity: we typically use a model for 1 time series, but there are situations that we need to use that model for multiple time series. We can even define some shared hyperpriors to infer some properties across time series.

Some references for batch gps: https://docs.gpytorch.ai/en/latest/examples/08_Advanced_Usage/index.html#batch-gps or https://forum.pyro.ai/t/gaussian-processes-fit-using-plates/1742?u=fehiepsi

I would recommend not supporting it if you think it is unnecessary or complicated. Personally, I don't use CAR so I don't know how people uses it in practice (but to me, batching is useful in general).

Reply:
Ahhh okay. Yes, for one of my models I am looking into using a GP with multiple outputs. I'm more used to defining them like [this](https://docs.gpytorch.ai/en/v1.5.1/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html).

I haven't seen people doing batch CARs before. When I have seen multiple CAR effects in models, including my own, I have seen them defined as two separate effects rather than one call.

Let's stick to no batching for now. Given that the CAR distribution in 1D can be used to specify a gaussian random walk, I can see people _might_ want to use it for multiple time series, so I'll put in a TODO: support batched W. And we can see if there is demand. But I think batched GP demand will be far more pressing than batched CAR

Reply:
Unless of course, you think batched W is hardly any more work...

Reply:
Yeah, I think it is unnecessary for now. We just need to left a message that it can be supported to encourage users to submit feature requests if they need. ;)